packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -249,26 +249,31 @@ export type Chunker<
   ChunkMetadata extends Jsonifiable = Jsonifiable
 > = (document: Document<DocumentMetadata>) => Promise<Chunk<ChunkMetadata>[]>;
 
-/** A simple token size based text chunker. This is a good starting point for text documents. */
-export const defaultChunker = async <Metadata extends Jsonifiable = Jsonifiable>(doc: Document<Metadata>) => {
-  const splitter = new TokenTextSplitter({
-    encodingName: 'gpt2',
-    chunkSize: 600,
-    chunkOverlap: 100,
-  });
-
-  const lcDocs = await splitter.createDocuments(doc.pageContent);
-  const chunks = lcDocs.map(
-    (lcDoc) =>
-      ({
-        content: lcDoc.pageContent,
-        documentName: doc.name,
-        metadata: doc.metadata,
-      } as Chunk<Metadata>)
-  );
+/** Create a chunker with the given parameters. */
+export function makeChunker(chunkSize: number, chunkOverlap: number): Chunker {
```

Comment:
Isn't this losing the metadata type?

Reply:
Yes, but I don't understand enough about TypeScript to know what the right thing to do here - can you suggest a fix?

Reply:
I'm certainly no TypeScript expert either, but I think the suggestion I just added should be right!

Reply:
I'll take a look at this after the PR lands. I don't want to hold the PR up on this.